File: solutions/dataStructures/hashTables/validAnagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        
        counter = {}

        for x in s:
            counter[x] = counter.get(x, 0) + 1
        
        for x in t:
            if x not in counter or counter[x] == 0:
                return False
            counter[x] -= 1
        
        return True

File: solutions/java/grind75/week1/validAnagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) return false;

        Map<Character, Integer> counter = new HashMap<>();

        for (char c : s.toCharArray()) {
            counter.put(c, counter.getOrDefault(c, 0) + 1);
        }

        for (char c : t.toCharArray()) {
            counter.put(c, counter.getOrDefault(c, 0) - 1);
        }

        for (int val : counter.values()) {
            if (val != 0) return false;
        }

        return true;
    }
}